# This File Holds the Parameters for the PID controllers for each joint motor
hdt_rlb:
  # Publish all joint states -----------------------------------
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50

  # Position Controllers ---------------------------------------
  joint1_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint1 # Pan of the whole robotic arm
    pid: {p: 100.0, i: 0.0, d: 10.0}
  joint2_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint2 # Tilt of the whole robotic arm
    pid: {p: 500.0, i: 0.0, d: 200.0}
  joint4_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint4 # Tilt of the robotic arm elbow
    pid: {p: 500.0, i: 0.0, d: 200.0}
  joint5_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint5 # Rotation of the robotic arm elbow
    pid: {p: 100.0, i: 0.0, d: 10.0}
  joint6_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint6 # Tilt of the robotic arm hand
    pid: {p: 100.0, i: 0.0, d: 10.0}
  joint7_position_controller:
    type: effort_controllers/JointPositionController
    joint: joint7 # Pan of the robotic arm hand
    pid: {p: 100.0, i: 0.0, d: 10.0}
